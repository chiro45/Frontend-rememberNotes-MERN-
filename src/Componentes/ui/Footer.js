import React from 'react'

export const Footer = () => {
  return (
    <div className='footer__container'>
        
      <div className='footer__container-title'>
      <span className='footer__title'>RememberNotes <i className="fa-solid fa-book"></i></span>
      </div>

    </div>
  )
}
