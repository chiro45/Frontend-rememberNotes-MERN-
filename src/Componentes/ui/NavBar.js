
export const NavBar = () => {
  return (
    <div className="navbar__container">
        <div className="navbar__name-container">
            <span className="navbar-name">Luciano <i className="fa-solid fa-user-pen"></i></span>
        </div>


        <div className="navbar__logout-container">
            <button className="navbar-logout">Salir <i className="fa-solid fa-arrow-right-to-bracket"></i></button>
        </div>




    </div>
  )
}
