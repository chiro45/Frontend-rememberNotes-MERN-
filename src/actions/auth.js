import Swal from "sweetalert2"
import { fetchConToken, fetchSinToken } from "../helpers/fetch"
import { types } from "../types/types"


export const startLogin = (email, passwd)=>{
    return async(dispatch)=>{
        
      const resp = await fetchSinToken('auth',{email, passwd}, 'POST')
      const body = await resp.json()
      
      
      if(body.ok){
        localStorage.setItem('token',body.token )
        localStorage.setItem('token-init-date', new Date().getTime() )
        dispatch(login({
            uid: body.uid,
            name: body.name
        }))
    }else{
        Swal.fire('Error', body.msg,'error')
    }

      

    }
} 


export const startRegister = (name,email, passwd)=>{
    return async(dispatch)=>{
        const resp = await fetchSinToken('auth/register',{name, email, passwd}, 'POST')
        const body = await resp.json()
        
        if(body.ok){
            localStorage.setItem('token',body.token )
            localStorage.setItem('token-init-date', new Date().getTime() )

            dispatch(login({
                uid: body.uid,
                name: body.name
            }))
            Swal.fire('Creado', "Usuario creado con exito",'success')
        }else{
            Swal.fire('Error', body.msg,'error')
      }
    }
}

export const startChecking = ()=>{
    return async(dispatch)=>{
        const resp = await fetchConToken('auth/renew')
        const body = await resp.json()
        if(body.ok){
            localStorage.setItem('token',body.token )
            localStorage.setItem('token-init-date', new Date().getTime() )

            dispatch(login({
                uid: body.uid,
                name: body.name
            }))
           
        }else{
            dispatch(chekingFinish())
      }
    }
    }


const chekingFinish = ()=>({
    type: types.authChekingFinish
})


const login = (user)=>({
    type: types.authlogin,
    payload: user
})