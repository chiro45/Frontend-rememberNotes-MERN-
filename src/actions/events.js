import Swal from "sweetalert2"
import { fetchConToken } from "../helpers/fetch"
import { prepareEvents } from "../helpers/prepareEvents"
import { types } from "../types/types"

export const startAddNew = (event)=>{

    return async(dispath, getState)=>{

        const {uid, name} = getState().authReducer
        
        try {
            const resp = await fetchConToken('events',event, 'POST')
            const body = await resp.json()
           
            if(body.ok){
                event.id = body.eventoGuardado.id;
                event.user = {
                   _id: uid, 
                    name
                }
                dispath(addNew(event))
            }
            
        } catch (error) {
            console.log(error);
            Swal.fire('error','No se ha guardado','error')
        }

        

    }    


}

const addNew = (event)=>({
    type: types.eventAddnew,
    payload: event    
});


export const clearActiveNote = ()=>({
    type: types.eventClearActiveEvent
})

export const setActive = (event)=>({
    type: types.eventSetActive,
    payload: event
})

export const startEventDelete = ()=>{
    return async(dispath, getState)=>{
        const {id} = getState().calendar.activeEvent
        try {
            const resp = await fetchConToken(`events/${id}`, {}, "DELETE")
            const body = await resp.json()
             if(body.ok){
                 dispath(eventDelete())
                 Swal.fire('success','Evento Elminado','success')
             }
            
           } catch (error) {
               console.log(error)
               Swal.fire('error', 'No se pudo Eliminar el evento', 'error')
           }
        }
    }



const eventDelete = ()=>({
    type: types.eventDelete
  
})




export const eventStartLoading = ()=>{

    return async(dispath)=>{

       try {
           const resp =  await fetchConToken('events')
           const body = await resp.json()

           const events =   prepareEvents( body.eventos)

            dispath(eventLoader(events))
          
       } catch (error) {
           console.log(error)
           Swal.fire('error','No se pudieron cargar los eventos','error')
       }


    }

}

const eventLoader = (eventos)=>({
    type : types.eventLoader,
    payload : eventos
})


export const startEventUpdated = (evento)=>{
    return async(dispath)=>{
       try {
        const resp = await fetchConToken(`events/${evento.id}`, evento, "PUT")
        const body = await resp.json()
         if(body.ok){
             dispath(eventUpdated(evento))
             Swal.fire('success','Evento Actualizado','success')
         }
        
       } catch (error) {
           console.log(error)
           Swal.fire('error', 'No se pudo actualizar el evento', 'error')
       }
    }
}
const eventUpdated = (evento) =>({
    type: types.eventUpdate,
    payload: evento
})
