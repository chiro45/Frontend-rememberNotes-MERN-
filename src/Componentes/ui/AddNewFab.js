
import { useDispatch } from 'react-redux'
import { uiOpenModal } from '../../actions/ui';

export const AddNewFab = () => {
    const dispatch = useDispatch();
    const handleClick=()=>{
        dispatch(uiOpenModal())
    }
  return (
    <button className='fabs' onClick={handleClick}><i className='fas fa-plus'/></button>
  )
}
