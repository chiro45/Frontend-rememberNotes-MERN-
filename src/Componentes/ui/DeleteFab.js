
import { useDispatch } from 'react-redux'
import { eventDelete } from '../../actions/events';


export const DeleteFab = () => {
    const dispatch = useDispatch();
    const handleClick=()=>{
        dispatch(eventDelete())
    }
  return (
    <button className='fabs delete' onClick={handleClick}><i className="fa-solid fa-trash-can"></i></button>
  )
}
