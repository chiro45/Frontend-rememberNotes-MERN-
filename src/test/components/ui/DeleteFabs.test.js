import React from 'react';

import { Provider } from 'react-redux';

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import '@testing-library/jest-dom'
import { DeleteFab } from '../../../Componentes/ui/DeleteFab';
import { mount } from 'enzyme';
import { startEventDelete } from '../../../actions/events';

jest.mock('../../../actions/events',()=>({
    startEventDelete: jest.fn()
}))
 
const middlewere = [thunk]

const mockStore = configureStore(middlewere)

const initState = {}

let store = mockStore(initState)
store.dispatch= jest.fn();

const wrapper = mount( 
    <Provider store={store} >
        <DeleteFab/>
    </Provider>
)



describe('test in <DeleteFab/>', () => {
    
    test('should to look correctly', () => { 

       
        expect(wrapper).toMatchSnapshot();
     })


     test('should call eventStartDelete when click the button', () => { 

        wrapper.find('button').prop('onClick')()

        expect(startEventDelete).toBeCalledWith()
     })




});
