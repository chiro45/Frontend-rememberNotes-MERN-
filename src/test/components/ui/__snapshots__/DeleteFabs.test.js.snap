// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <DeleteFab/> should to look correctly 1`] = `
<button
  className="fabs delete"
  onClick={[Function]}
>
  <i
    className="fa-solid fa-trash-can"
  />
</button>
`;
